### PR TITLE
Fix/undo on menu open

### LINKED ISSUE #411: [bug] Undo doesnt work when the detail popup window is open
To reproduce:
- Put some shapes on the canvas and select them
- Open the detail menu (the one labeled '...')
- Click any action in there, like align the shapes etc
- Do 'undo' using the keyboard shortcut (cmd/ctrl z)
- Note that the action does NOT undo. Undo does work if triggered via Hamburger -> Edit -> Undo

Found on: https://www.tldraw.com/

Example:

https://user-images.githubusercontent.com/3643611/143810341-8c62d9e1-2f6f-4bad-b646-a8ace8ca320e.mov

### PR DESCRIPTION
**Pull request by megnakokkalera**
Resolves #411 
Not sure where the ready for review template is
But pretty much created a boolean function that returns true if the menu is toggled open
Then in canHandleEvent in KeyboardShortcuts, if the menu is opened, this function returns true allowing you to undo when the action menu is open

### CODE REVIEW

**Comment by megnakokkalera** [comment]
video of the change is located at #411

File: package.json [hunk 1]
```diff
@@ -65,16 +65,16 @@
     "lerna": "^3.22.1",
     "lint-staged": "^11.2.6",
     "prettier": "^2.4.1",
+    "pretty-quick": "^3.1.2",
```
Comment (proful) [inline package.json:68]: I don't think any changes required for this package json file

**Comment by proful** [comment]
yarn lock file need not be committed for this PR

**Comment by megnakokkalera** [comment]
> yarn lock file need not be committed for this PR

youre right for some reason I couldn't get it to not commit. i can try recommitting again if the changes are okay?

**Comment by steveruizok** [comment]
Hey, works great! A few changes:

- Rather than toggling the menu, let's set it instead. Toggles can get into weird states too easily.
- Removed the `isStyleOpen` property from the model
- Added a flag to allow keyboard shortcuts when the menu is open

There are some keyboard shortcuts (such as the arrow keys) that conflict with keyboard navigation. For example, a user wouldn't want to be moving their selected shapes while moving within the menu. So I've added a flag for this to only allow shortcuts to skip the menu if that flag is explicitly set. 

Feel free to tweak some of the shortcuts to work / not work when the menu is open. You might also flip the default, so that it allows shortcuts unless explicitly set not to do so. (Maybe it's just the arrows that need to be blocked when the menu is open?)

**Comment by megnakokkalera** [comment]
The changes are great, thank you!
Would it be possible to accept this current PR and I can make another PR to tweak the shortcuts? Just wanted to try and fix the undo button not working for this current PR and I can make subsequent changes elsewhere.

**Comment by steveruizok** [comment]
Sounds good. Thanks again @megnakokkalera !

**Pull request merged**